cmd/kusk/cmd/deploy.go

```diff
@@ -107,63 +108,68 @@ var deployCmd = &cobra.Command{
 				if err := k8sclient.Update(ctx, api, &client.UpdateOptions{}); err != nil {
 					return err
 				}
+				fmt.Printf("api.gateway.kusk.io/%s updated\n", api.Name)
 			} else {
 				return err
 			}
 		} else {
 			fmt.Printf("api.gateway.kusk.io/%s created\n", api.Name)
 		}
 
-		if watch {
-			var watcher *filewatcher.FileWatcher
-			absoluteApiSpecPath := file
-			if err != nil {
-				return err
-			}
+		if _, e := url.ParseRequestURI(file); e != nil {
+			if watch {
+				var watcher *filewatcher.FileWatcher
+				absoluteApiSpecPath := file
+				if err != nil {
+					return err
+				}
 
-			watcher, err = filewatcher.New(absoluteApiSpecPath)
-			if err != nil {
-				return err
-			}
-			defer watcher.Close()
-
-			done := make(chan os.Signal, 1)
-			signal.Notify(done, syscall.SIGINT, syscall.SIGTERM)
-
-			if watcher != nil {
-				ui.Info(ui.White("⏳ watching for API changes in " + file))
-				go watcher.Watch(func() {
-					ui.Info("✍️ change detected in " + file)
-					manifest, err := getParsedAndValidatedOpenAPISpec(file)
-					if err != nil {
-						ui.Fail(err)
-					}
-					api := &v1alpha1.API{}
-					if err := yaml.Unmarshal([]byte(manifest), api); err != nil {
-						ui.Info("there has been an error in API manifest", err.Error())
-					}
-
-					if len(api.Namespace) == 0 {
-						api.Namespace = "default"
-					}
-					if len(api.Name) == 0 {
-						api.Name = name
-					}
-					ap := &v1alpha1.API{}
-					if err := k8sclient.Get(ctx, client.ObjectKey{Namespace: api.Namespace, Name: api.Name}, ap); err != nil {
-						fmt.Fprintln(os.Stderr, err)
-					}
-					api.SetResourceVersion(ap.GetResourceVersion())
-
-					if err := k8sclient.Update(ctx, api, &client.UpdateOptions{}); err != nil {
-						fmt.Fprintln(os.Stderr, err)
-					} else {
-						fmt.Printf("api.gateway.kusk.io/%s updated\n", api.Name)
-					}
-
-				}, done)
+				watcher, err = filewatcher.New(absoluteApiSpecPath)
+				if err != nil {
+					return err
+				}
+				defer watcher.Close()
+
+				done := make(chan os.Signal, 1)
+				signal.Notify(done, syscall.SIGINT, syscall.SIGTERM)
+
+				if watcher != nil {
+					ui.Info(ui.White("⏳ watching for API changes in " + file))
+					go watcher.Watch(func() {
+						ui.Info("✍️ change detected in " + file)
+						manifest, err := getParsedAndValidatedOpenAPISpec(file)
+						if err != nil {
+							ui.Fail(err)
+						}
+						api := &v1alpha1.API{}
+						if err := yaml.Unmarshal([]byte(manifest), api); err != nil {
+							ui.Info("there has been an error in API manifest", err.Error())
+						}
+
+						if len(api.Namespace) == 0 {
```

Comment:
Is this intentional?

1. Assume `yaml.Unmarshal([]byte(manifest), api)` returns an error.
2. The `api` variable is accessed again further on.

Reply:
The error check above is pointless as the manifest is generated in `getParsedAndValidatedOpenAPISpec` so if that fails it won't get to this point. At this point API instance will be available.

Reply:
for example if logically the openapi manifest doesn't pass you get this message:
```
# go run cmd/kusk/main.go deploy -i example.yaml -w
Error: OperationFinalSubOptions: (GET/test: (upstream: (service: (port: cannot be blank.).).); POST/another: (upstream: (service: (port: cannot be blank.).).).).
Error: OperationFinalSubOptions: (GET/test: (upstream: (service: (port: cannot be blank.).).); POST/another: (upstream: (service: (port: cannot be blank.).).).).
exit status 1
```
or 
```
# go run cmd/kusk/main.go deploy -i example.yaml -w
api.gateway.kusk.io/simple-apis updated
⏳ watching for API changes in example.yaml
✍️ change detected in example.yaml
⨯ no upstream or mocking configuration found for path POST/something 💔
```